### PR TITLE
Tpwdcee 6604 feature

### PR DESCRIPTION
**Pull request by marko-babic**

### CODE REVIEW

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/wirecardpaymentgateway.php  1
         

Complexity increasing per file
==============================
- wirecardpaymentgateway/wirecardpaymentgateway.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5631568&bid=18444440)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/helper/CurrencyHelper.php  2
+ wirecardpaymentgateway/classes/Service/OrderAmountCalculatorService.php  1
+ wirecardpaymentgateway/helper/TransactionBuilder.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  2
+ wirecardpaymentgateway/classes/Response/Failure.php  1
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/models/Transaction.php  -1
- wirecardpaymentgateway/helper/OrderManager.php  -1
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -1
- wirecardpaymentgateway/classes/Service/OrderStateManagerService.php  -9
         

Complexity increasing per file
==============================
- wirecardpaymentgateway/wirecardpaymentgateway.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5631568&bid=18444440)

File: wirecardpaymentgateway/wirecardpaymentgateway.php [hunk 1]
```diff
@@ -634,6 +634,13 @@ private function createInputFields($radioType, $radioOptions)
                         }
                         break;
 
+                    case 'hint':
+                        if (isset($elem["desc"])) {
+                            $style = "font-style: initial; color: initial; font-size: 15px;";
+                            $elem["desc"] = "<span style=\"$style\">" . html_entity_decode($elem["desc"]) . "</span>";
```
Comment (sebastijansakac) [inline wirecardpaymentgateway/wirecardpaymentgateway.php:640]: Please use Tools::htmlentitiesDecodeUTF8() from prestashop

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wirecardpaymentgateway/wirecardpaymentgateway.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5631568&bid=18444440)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/classes/Service/OrderAmountCalculatorService.php  1
+ wirecardpaymentgateway/helper/CurrencyHelper.php  2
+ wirecardpaymentgateway/helper/TransactionBuilder.php  1
+ wirecardpaymentgateway/classes/Response/Failure.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  2
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -1
- wirecardpaymentgateway/helper/OrderManager.php  -1
- wirecardpaymentgateway/classes/Service/OrderStateManagerService.php  -9
- wirecardpaymentgateway/models/Transaction.php  -1
         

Complexity increasing per file
==============================
- wirecardpaymentgateway/wirecardpaymentgateway.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5631568&bid=18444440)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -2
- wirecardpaymentgateway/models/Transaction.php  -1
         

Complexity increasing per file
==============================
- wirecardpaymentgateway/wirecardpaymentgateway.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5631568&bid=18444440)

**Pull request merged**